t/load_csv_or_excel_file.t

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/perl -w
+
+use strict;
+use warnings;
+use utf8;
+
+use Test::More;
+use Log::Any::Adapter 'TAP';
+
+use File::Basename "dirname";
+
+use ProductOpener::Producers qw(load_csv_or_excel_file);
+
+my $inputs_dir = dirname(__FILE__) . "/inputs/";
+
+
+# Ademe excel test
+my $results_ref = load_csv_or_excel_file($inputs_dir . "eco-score-template.xlsx");
+my @expected_headers = (
+    "EAN", "Nom", "Marque", "Catégorie", "Boisson",
+    "Label 1", "Label 2",
+    "Ingrédient", "Origine", "Pourcentage",
+    "Ingrédient - 2", "Origine - 2", "Pourcentage - 2",
+    "Ingrédient - 3", "Origine - 3", "Pourcentage - 3",
+    "Ingrédient - 4", "Origine - 4", "Pourcentage - 4",
+    "Ingrédient - 5", "Origine - 5", "Pourcentage - 5",
+    "Ingrédient - 6", "Origine - 6", "Pourcentage - 6",
+    "Format", "Matériau",
+    "Format - 2", "Matériau - 2",
+    "Format - 3", "Matériau - 3",
+    "Huile de palme", "Poissons menacés"
+);
+
+# we have headers
+is_deeply(
+    \@{$results_ref->{headers}},
```

Comment:
```suggestion
    $results_ref->{headers},
```

@{$array_ref} dereferences $array_ref to make it an array, and then \ makes it a reference again.